src/autopick/autopick-drawer.cpp

```diff
@@ -66,7 +66,9 @@ void draw_text_editor(PlayerType *player_ptr, text_body_type *tb)
 {
     int by1 = 0, by2 = 0;
 
-    term_get_size(&tb->wid, &tb->hgt);
+    const auto [wid, hgt] = term_get_size();
+    tb->wid = wid;
+    tb->hgt = hgt;
```

Comment:
この部分のように、すぐに別の変数に入れ直すためだけに一時変数で受けている場所がいくつかありますが、std::tie を使うことで一時変数を削除できます。（要tupleヘッダ）

```suggestion
    std::tie(tb->wid, tb->hgt) = term_get_size();
```

Reply:
VSは文句言ってこなかったのでC++はそんなもんかと思いましたがやっぱりダメなんですね
直しておきます
(C#は\_ 1文字でdiscardなのにぐぬぬ)

Reply:
ちなみに上で紹介している std::tie は C++17 で構造化束縛が導入される前に構造化束縛ライクなものとして使用されていたのですが、読み捨てたい場合は std::ignore というものがあり `std::tie(std::ignore, hgt) = ...;` のようにすれば実現できます。
そこまで分かっておきながら、構造化束縛の導入時に読み捨てる文法を入れなかったのは謎です。

Reply:
これは気持ち悪い (シグネチャ変更でせっかく1行にしたのに2行のまま)のでmaybe unused を使うことにします
```cpp
int hgt;
std::tie(std::ignore, hgt) = get_screen_size();
```